### DISCUSSION TITLE
Improve debugging info in useAtomsDevtools

### DISCUSSION THREAD

**Discussion by dai-shi** [jotaijs/jotai-devtools#55] [General]
Moved from https://github.com/pmndrs/jotai/issues/931

Experimental PR: https://github.com/pmndrs/jotai/pull/943

If we do something, it will be `jotai`, not only `jotai-devtools`, but this should be a good place for discussion.

**Comment by dai-shi** [comment]
This can be a hint: https://github.com/pmndrs/zustand/discussions/2061#discussioncomment-7109605 by @AlexeevSasha

**Comment by Wendystraite** [comment]
I quite like Zedux's way of capturing the React component name without a stack trace.

Basically using React's internal API.

Snippet taken from https://github.com/Omnistac/zedux/blob/master/packages/react/src/hooks/useReactComponentId.ts : 

```ts
const react19KeyBase = '_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE';
const clientKey = `__CLIENT${react19KeyBase}` as const;
const serverKey = `__SERVER${react19KeyBase}` as const;

export function useReactComponentId() {
  const component: MaybeComponent =
    ((React as React19)[clientKey] ?? (React as React19)[serverKey])?.A?.getOwner?.().type ??
    (React as React18).__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED?.ReactCurrentOwner?.current?.type;
  const name = component?.displayName || component?.name || 'unknown';
  const id = useId();
  return `${name}-${id}`;
}
```

Cons : 
- It can break after any React update
- It doesn't give you the potential names of custom hooks, only the React component using them
- (as I can see) It must be called in the render function, not inside an useEffect, callback or whatever. This means that Jotai core must be changed in order to get the component's name.

**Comment by Wendystraite** [comment]
Good news, React 19.1 introduced [captureOwnerStack](https://react.dev/reference/react/captureOwnerStack) : an API to retrieve the list of parent components.

I've been toying with it a bit and released [an experimental API for jotai-logger](https://github.com/Wendystraite/jotai-logger/blob/main/README.md#component-tracking-experimental) today using both `captureOwnerStack` and React 19's internal method `getOwner`.

Instead of trying to parse error strack traces ourselves, this is much more stable.
